src/dkpy/uncertainty_structure.py

```diff
@@ -0,0 +1,200 @@
+        """Get number of output of the uncertainty block."""
+        return self._num_outputs
+
+    @property
+    def is_diagonal(self):
+        """Get boolean for diagonal uncertainty."""
+        return self._is_diagonal
+
+    @property
+    def is_complex(self):
+        """Get boolean for complex uncertainty."""
+        return self._is_complex
+
+
+class RealDiagonalBlock(UncertaintyBlock):
+    """Real-valued diagonal uncertainty block."""
+
+    def __init__(self, num_channels: int):
+        """Instantiate :class:`RealDiagonalBlock`.
+
+        Parameters
+        ----------
+        num_channels : int
+            Number of inputs/outputs to the uncertainty block.
+
+        Raises
+        ------
+        ValueError
+            If ``num_channels`` is not greater than zero.
+        """
+        if num_channels <= 0:
+            raise ValueError("`num_channels` must be greater than 0.")
+
+        super().__init__(
+            num_inputs=num_channels,
+            num_outputs=num_channels,
+            is_diagonal=True,
+            is_complex=False,
+        )
+
+
+class ComplexDiagonalBlock(UncertaintyBlock):
+    """Complex-valued diagonal uncertainty block."""
+
+    def __init__(self, num_channels: int):
+        """Instantiate :class:`ComplexDiagonalBlock`.
+
+        Parameters
+        ----------
+        num_channels : int
+            Number of inputs/outputs to the uncertainty block.
+
+        Raises
+        ------
+        ValueError
+            If ``num_channels`` is not greater than zero.
+        """
+        if num_channels <= 0:
+            raise ValueError("`num_channels` must be greater than 0.")
+
+        super().__init__(
+            num_inputs=num_channels,
+            num_outputs=num_channels,
+            is_diagonal=True,
+            is_complex=True,
+        )
+
+
+class ComplexFullBlock(UncertaintyBlock):
+    """Complex-valued full uncertainty block."""
+
+    def __init__(self, num_inputs: int, num_outputs: int):
+        """Instantiate :class:`ComplexFullBlock`.
+
+        Parameters
+        ----------
+        num_inputs : int
+            Number of inputs (columns) to the uncertainty block.
+        num_outputs : int
+            Number of outputs (rows) to the uncertainty block.
+
+        Raises
+        ------
+        ValueError
+            If ``num_inputs`` is not greater than zero.
+        ValueError
+            If ``num_outputs`` is not greater than zero.
+        """
+
+        if num_inputs <= 0:
+            raise ValueError("`num_inputs` must be greater than 0.")
+        if num_outputs <= 0:
+            raise ValueError("`num_outputs` must be greater than 0.")
+
+        super().__init__(
+            num_inputs=num_inputs,
+            num_outputs=num_outputs,
+            is_diagonal=False,
+            is_complex=True,
+        )
+
+
+def _convert_matlab_block_structure(
+    block_structure_matlab: np.ndarray,
+) -> Sequence[UncertaintyBlock]:
+    """
```

Comment:
Still needs a one-liner description on the line with the `"""`, then a line break, then the big doc